src/dryrun.ts

```diff
@@ -167,3 +167,159 @@ export async function createDryrun({
     sources,
   });
 }
+
+class DryrunStackValue {
+  type: number = 0;
+  bytes: string = '';
+  uint: number = 0;
+
+  constructor(sv: Record<string, any>) {
+    this.type = sv.type;
+    this.bytes = sv.bytes;
+    this.uint = sv.uint;
+  }
+
+  toString(): string {
+    if (this.type === 1) {
+      return this.bytes;
+    }
+    return this.uint.toString();
+  }
+}
+
+class DryrunTraceLine {
+  line: number = 0;
+  pc: number = 0;
+  stack: DryrunStackValue[] = [];
+
+  constructor(line: Record<string, any>) {
+    this.line = line.line;
+    this.pc = line.pc;
+    this.stack = line.stack.map(
+      (sv: Record<string, any>) => new DryrunStackValue(sv)
+    );
+  }
+
+  traceLine(): [number, number, string] {
+    return [
+      this.line,
+      this.pc,
+      `[${this.stack.map((sv) => sv.toString()).join(',')}]`,
+    ];
+  }
+}
+
+class DryrunTrace {
+  trace: DryrunTraceLine[] = [];
+
+  constructor(t: Record<string, any>[]) {
+    if (t === undefined) return;
+    this.trace = t.map((line) => new DryrunTraceLine(line));
+  }
+
+  getTrace(): any[] {
```

Comment:
This returns `Array<[number, number, string]>`?

Btw I would suggest making an object interface for that tuple instead of using `[number, number, string]`. Something like this would be much clearer:

```typescript
interface Trace { // or whatever name makes sense
    line: number,
    pc: number,
    stack: string,
}
```